daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/BinaryNumberTraits.scala

```diff
@@ -165,3 +165,21 @@ trait PrefixedLengthParserMixin {
     }
   }
 }
+
+/**
+ * This mixin doesn't require parsing the prefix length element and just uses
+ * the state's bitLimit and position to get the bitLength instead
+ */
+trait PrefixedLengthParserMixin2 {
```

Comment:
Need a different name for this. Numbers are not descriptive enough. Maybe we renames these something like `PrefixLengthFromParserMixin` and `PrefixLengthFromBitLimitMixin`? Something to make it more clear how they differ without having to read the code.